src/main/java/com/ldtteam/structurize/placement/handlers/placement/PlacementHandlers.java

```diff
@@ -249,13 +252,23 @@ public List<ItemStack> getRequiredItems(
                     posBelow = posBelow.below();
                     final boolean isFirstTest = i == 0;
                     final BlockState possibleSupport = BlockUtils.getWorldgenBlock(world, posBelow, bp -> isFirstTest ? blockState : null);
-                    if (possibleSupport != null && BlockUtils.canBlockFloatInAir(possibleSupport) && !canHandle(world, posBelow, possibleSupport))
+                    if (possibleSupport != null && BlockUtils.isAnySolid(possibleSupport) && BlockUtils.canBlockFloatInAir(possibleSupport) && !canHandle(world,
+                        posBelow,
+                        possibleSupport))
                     {
                         supportBlockState = possibleSupport;
```

Comment:
imho weird, the old code already guarantees: possibleSupport is either DIRT or !canHandle(...) => it cannot be floating, so i'm now assuming that DIRT somehow appeared in FallingBlock or Fallable due to mixin stuff? or am i missing sth?

Reply:
A few mods (e.g. terrafirmacraft) change dirt to be fallable, so that could be possible

Reply:
Yeah, so "proper" fix for these mods would be to have more defaults possible, like stone. or dimension default block assuming it's non fallable

Reply:
I posted the original bug report, is there anything I can test or do to assist with finding a possible cause or contributing mod? I'm using a quite extensive modpack and so far all works well apart from this item. If it is useful or make any difference; the error occurs when opening the 'Build Options' menu, sometimes the menu opens but freezes shortly after, sometimes it doesn't open at all - maybe coincidental?

Reply:
ye its a fallback for mods doing weird stuff or using weird blocks on worldgen surface